CHANGELOG.md

```diff
@@ -10,6 +10,7 @@
 
 ### Fixes
 
+* **Partitioning raises on file-like object with `.name` not a local file path.** Partitioning a file using the `file=` argument with a file-like object (e.g. io.BytesIO) having a `.name` attribute with a value not a path to a file present on the local filesystem raises `FileNotFoundError`. This prevents setting the `.name` attribute to give a `.metadata.filename` value to elements formed from for example a file retrieved from a network location via HTTP.
```

Comment:
This is a little hard to follow. If I'm following correctly, previously `FileNotFoundError` was raised when a file-like object with a `.name` attribute (that was not a path) was partitioned using `file=`, right?

Reply:
What you said in the PR description is clearer:
This fix allows a bytes payload to be given a filename in its `metadata` even though such a file never existed on the local filesystem. This fix explicitly checks for the existence of a file at the given path before accessing it to get its modified date. Return None (already a legitimate return value) when no such file exists.

Reply:
Yeah, the triggering conditions were these:
1. You call partition_{x}() (pretty much any of them).
2. You use the `file` arg to provide the input document.
3. That file value is a file-like object like `io.BytesIO`, not the `io.BufferedReader` you get back from `with open(path) as f`.
4. You assign a name to the file-like object using `file.name = meaningful_name`. You do that so it shows up in the `metadata.filename` field for each element. Maybe you downloaded the bytes of the document over the network so it doesn't have a path on the filesystem, but you still want to indicate what document the elements came from.

Then when the partitioner went to compute the last-modified date for metadata it would try to access the file at `file.name` to get its modified date without checking whether it actually existed on the filesystem or not.